website/client-old/js/services/taskServices.js

```diff
@@ -277,11 +277,100 @@ angular.module('habitrpg')
         modalScope.removeTask = scopeInc.removeTask;
       }
       modalScope.cancelTaskEdit = cancelTaskEdit;
+      modalScope.summary = generateSummary(modalScope.task);
 
-      $rootScope.openModal('task-edit', {scope: modalScope })
-        .result.catch(function() {
-          cancelTaskEdit(task);
-        });
+      modalScope.task._edit.repeatsOn = 'dayOfMonth';
+      if (modalScope.task._edit.weeksOfMonth.length > 0) {
+        modalScope.task._edit.repeatsOn = 'dayOfWeek';
+      }
+
+      $rootScope.openModal('task-edit', {
+        scope: modalScope,
+        controller: function ($scope) {
+          $scope.$watch('task._edit', function (newValue, oldValue) {
+            $scope.summary = generateSummary(modalScope.task);
+
+            if ($scope.task._edit.repeatsOn == 'dayOfMonth') {
+              var date = moment().date();
+              $scope.task._edit.weeksOfMonth = [];
+              $scope.task._edit.dayOfMonth = [date]; // @TODO This can handle multiple dates later
+            } else if ($scope.task._edit.repeatsOn == 'dayOfWeek') {
+              var week = Math.ceil(moment().date() / 7) - 1;
+              var dayOfWeek = moment().day();
+              var shortDay = numberToShortDay[dayOfWeek];
+              $scope.task._edit.dayOfMonth = [];
+              $scope.task._edit.weeksOfMonth = [week]; // @TODO: This can handle multiple weeks
+              for (var key in $scope.task._edit.repeat) {
+                $scope.task._edit.repeat[key] = false;
+              }
+              $scope.task._edit.repeat[shortDay] = true;
+            }
+          }, true);
+        },
+      })
+      .result.catch(function() {
+        cancelTaskEdit(task);
+      });
+    }
+
+    /*
+     * Summary
+     */
+
+    var frequencyMap = {
+      'daily': 'days',
+      'weekly': 'weeks',
+      'monthly': 'months',
+      'yearly': 'years',
+    };
+
+    var shortDayToLongDayMap = {
+      'su': 'Sunday',
```

Comment:
can we use i18n strings here? moment.locale should be available in the client https://momentjs.com/docs/#/i18n/